src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -150,13 +150,25 @@ public static class PrereleaseCompatibilityTransformer
 
         private static bool ApplyChangesFromTC35(JObject sarifLog)
         {
+            UpdateSarifLogVersionAndSchema(sarifLog);
+
+            bool modifiedLog = false;
+            
             // https://github.com/oasis-tcs/sarif-spec/issues/366
-            ConvertAllToolComponentArtifactIndicesToArtifactLocations(sarifLog);
+            modifiedLog |= ConvertAllToolComponentArtifactIndicesToArtifactLocations(sarifLog);
 
-            return true;
+            if (sarifLog["runs"] is JArray runs)
+            {
+                foreach (JObject run in runs)
+                {
+                    modifiedLog |= RenameProperty(run, "defaultFileEncoding", "defaultEncoding");
+                }
+            }
+
```

Comment:
this should go in its own method, we have maintained a clean readability for all our "ApplyChanges.." methods to quickly refer to the changes with descriptive method names.